File: jj/code_9.17/annotation-app/backend/app/schemas/recommendation.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class RecommendationResponse(BaseModel):
    window_id: int
    status: Literal["pending", "success", "failed"]
    recommended_label: Literal["normal", "abnormal"] | None
    recommended_anomaly_type: str | None
    reason: str | None
    model: str | None
    error_message: str | None
    pending_suggestion_id: int | None = None
    created_at: datetime
    updated_at: datetime


class RecommendationBatchProgressResponse(BaseModel):
    task_id: int
    total_windows: int
    success_count: int
    failed_count: int
    pending_count: int
    not_started_count: int


class WindowMultiAnalysisFileItem(BaseModel):
    source_log_file_id: int
    logical_path: str
    label: Literal["normal", "abnormal"]
    anomaly_type: str | None = None
    reason: str | None = None


class WindowMultiAnalysisResponse(BaseModel):
    window_id: int
    status: Literal["success", "failed"]
    multiple_faults: bool
    suggest_subdivide: bool
    suggested_window_seconds: int | None = None
    files: list[WindowMultiAnalysisFileItem]
    model: str | None = None
    error_message: str | None = None

File: jj/code_9.24/annotation-app/frontend/src/types/recommendation.ts
export type RecommendationStatus = 'pending' | 'success' | 'failed'

export interface Recommendation {
  window_id: number
  status: RecommendationStatus
  recommended_label: 'normal' | 'abnormal' | null
  recommended_anomaly_type: string | null
  reason: string | null
  model: string | null
  error_message: string | null
  pending_suggestion_id: number | null
  created_at: string
  updated_at: string
}

export interface RecommendationBatchProgress {
  task_id: number
  total_windows: number
  success_count: number
  failed_count: number
  pending_count: number
  not_started_count: number
}

export interface WindowMultiAnalysisFileItem {
  source_log_file_id: number
  logical_path: string
  label: 'normal' | 'abnormal'
  anomaly_type: string | null
  reason: string | null
}

export interface WindowMultiAnalysis {
  window_id: number
  status: 'success' | 'failed'
  multiple_faults: boolean
  suggest_subdivide: boolean
  suggested_window_seconds: number | null
  files: WindowMultiAnalysisFileItem[]
  model: string | null
  error_message: string | null
}
